packages/grafana-ui/src/components/Table/JSONViewCell.tsx

```diff
@@ -29,32 +29,38 @@ export const JSONViewCell: FC<TableCellProps> = (props) => {
   const content = <JSONTooltip value={value} />;
 
   return (
-    <div {...cellProps} className={tableStyles.cellContainer}>
-      <Tooltip placement="auto" content={content} theme="info-alt">
+    <Tooltip placement="auto-start" content={content} theme="info-alt">
+      <div {...cellProps} className={tableStyles.cellContainer}>
         <div className={cx(tableStyles.cellText, txt)}>{displayValue}</div>
-      </Tooltip>
-    </div>
+      </div>
+    </Tooltip>
   );
-};
+}
 
 interface PopupProps {
   value: any;
 }
 
-const JSONTooltip: FC<PopupProps> = (props) => {
-  const styles = useStyles((theme: GrafanaTheme) => {
-    return {
-      container: css`
-        padding: ${theme.spacing.xs};
-      `,
-    };
-  });
-
+function JSONTooltip(props: PopupProps): JSX.Element {
+  const styles = useStyles2(getStyles);
   return (
     <div className={styles.container}>
       <div>
-        <JSONFormatter json={props.value} open={4} />
+        <JSONFormatter json={props.value} open={4} className={styles.json} />
       </div>
     </div>
   );
-};
+}
+
+function getStyles(theme: GrafanaTheme2) {
+  return {
+    container: css`
+      padding: ${theme.spacing(0.5)};
+    `,
+    json: css`
+      max-width: fit-content;
```

Comment:
🧙

Reply:
Huh, what does that emoji mean?

Reply:
It means that you're a wizard Hugo! ⚡

Reply:
![](https://media0.giphy.com/media/3BKf0I2PVxAfC/giphy.gif)